File: urbanairship-sdk/src/main/java/com/urbanairship/util/UAMathUtil.java
/* Copyright 2017 Urban Airship and Contributors */

package com.urbanairship.util;

/**
 * A class containing basic math operations.
 */
public abstract class UAMathUtil {

    /**
     * Constrains an int to a min and max.
     *
     * @param value Value to clamp.
     * @param min The floor of the value.
     * @param max The ceiling of the value.
     * @return The value constrained between the min and max.
     */
    public static int constrain(int value, int min, int max) {
        if (value > max) {
            return max;
        }

        if (value < min) {
            return min;
        }

        return value;
    }
}

File: urbanairship-core/src/main/java/com/urbanairship/util/UAMathUtil.kt
/* Copyright Airship and Contributors */
package com.urbanairship.util

import androidx.annotation.RestrictTo
import kotlin.math.max

/**
 * A class containing basic math operations.
 *
 * @hide
 */
@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
public object UAMathUtil {

    /**
     * Constrains an int to a min and max.
     *
     * @param value Value to clamp.
     * @param min The floor of the value.
     * @param max The ceiling of the value.
     * @return The value constrained between the min and max.
     */
    public fun constrain(value: Int, min: Int, max: Int): Int {
        return value.coerceIn(min, max)
    }
}
